Add #-index-name: metadata to the tables.

DaveMielke opened the pull request:
Not done yet: fi-fi-8dot.ctb, no-no-8dot-fallback-6dot-g0.utb

bertfrees commented:
Some background for Christian: the idea is to have a new metadata field for selecting a table from a list fast and efficiently. This in contrast to the existing "display-name" field which is for describing a table accurately and nicely.

bertfrees commented:
I'll merge the Finnish table separately.

> AFAIK they use this in NVDA

They sure do.

I'll take care of the NEWS and everything. I also have suggestions for some changes coming up.

DaveMielke commented:
[quoted lines by Christian Egli on 2019/08/15 at 02:26 -0700]

Hi:

>However why did you at the same time remove a finish table? 

They were so identical that there wasn't any obvious way for the metadata to
describe them as being different from one another. The one with the extra
character was the one that was retained. That made only one character different
between them, and the one with what seemed to be the more accurate rendering is
in the one that was retained.

By the way, I did ask first.

>OK you explain why and I guess I'm fine with the resoning. Just next time it
might be better to put this in a separate PR

Yes, I understand that. I'm sorry.

>Also we need to mention the removal of tables in the NEWS file. AFAIK they use this in NVDA

I can put it back if you'd like. Another appraoch would be to put the old file
back but change it so that it includes the retained version, and then have the
metadata discover fi.tbl.

Does NVDA use fi.tbl or fi-fi.ctb?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
> Does NVDA use fi.tbl or fi-fi.ctb?

We don't know, we only know that they use our release notes to do changes in their code if needed.

egli commented:
> I can put it back if you'd like. Another approach would be to put the old file back but change it so that it includes the retained version, and then have the metadata discover fi.tbl.

Oh no, if it really is identical another table as you describe then good thing you discovered this and good riddance of this duplicated table.

We just need to mention the removal in the NEWS file, that's all.

DaveMielke commented:
Hi:

Shall I at least put fi.tbl back and move the metadata to it?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
No, don't. It's perfect like it is now. And I have the NEWS file covered.

bertfrees commented:
Finnish part merged in 78983d6

bertfrees commented:
@DaveMielke I have some suggestions (I have made these changes locally, so no need for you to change anything, just let me know what you think):

- Don't mention "forward" in either index-name or display-name. This is something that the user should never have to worry about. The program is always interested in either only forward translation, only backward translation, or both, so it should already filter out the unsuitable tables before presenting the list to the user. So, for example in da-dk-g16-lit.ctb, "Danish, uncontracted, 6-dot, forward" becomes "Danish, uncontracted, 6-dot" (index-name), and "Danish 6-dot uncontracted braille for forward translation" becomes "Danish 6-dot uncontracted braille" (display-name).
- You removed the "fully" from "fully contracted braille" (see for example da-dk-g26.ctb). I think I understand why you did this, but for me "contracted braille" also includes "partially contracted braille". I had added the "fully" to make it unambiguous. So my suggestion is to change it back to how it was before, and also add "fully" to the index-name. What do you think?
- In index-name (see for example da-dk-g26l.ctb) we could also maybe replace "partially contracted" with "contracted, partially"? Not sure about this though.
- I was thinking about moving the "(UEB)" in "Unified English (UEB) uncontracted braille" to the end, because I find it a bit awkward in the middle. So it would become "Unified English uncontracted braille (UEB)". Same for EBAE. Moreover, if it's at the end it doesn't disturb as much, so we could possibly add similar abbreviations of braille codes and authorities elsewhere (BAUK, BFU, UIB, etc.). Or maybe this is going to create too much noise, I dunno. But in any case, it should be consistent. For example, adding it for EBAE also justifies adding it for BAUK.
- Just a tiny detail, but I would change "Esperanto, x-system" to "Esperanto x-system" (without the comma), because the "x-system" is part of the original writing system (with language code "eo-xsistemo") rather than a variant of the braille code. I know, very subtle difference, but still.
- For the French tables we could drop the "unified" because there are no French tables that are not "unified". Also I'm not sure whether it has the same meaning as the "U" in UEB, because then why are there separate tables for computer braille?
- I would change "International Greek braille" to "Greek international braille" because the "international" has more affinity with "braille" then with "Greek".
- I would change the index-name of no-no-8dot-fallback-6dot-g0.utb, "Norwegian, uncontracted, 8-dot, with 6-dot fallback", to "Norwegian, uncontracted, 8-dot, 6-dot fallback" which makes it slightly more compact which is important for index-name.
- You changed the display-name of all the tables for Mandarin to have the word "Mandarin" in it. While I think this is the right thing to do for index-name, it's not necessarily so for display-name. For example "Taiwanese Mandarin braille" just doesn't sound as good as "Taiwanese braille", and the latter is what is also used on the Wikipedia page. Wikipedia also talks about "(Mainland) Chinese braille", not "Mainland Mandarin braille". Besides "Mainland" without "China" doesn't even make that much sense, so I also suggest to replace "mainland" with "mainland China" in all the index-names. And for consistency in the index-names, I would also add "mainland China" to the index-name of the "2-cell" table. In summary, these are the names for the Mandarin tables I'm proposing:

   index-name:
   - Mandarin, mainland China, without tone
   - Mandarin, mainland China, with tone
   - Mandarin, mainland China, 2-cell
   - Mandarin, Taiwan
   
   display-name:
   - Chinese braille without tone, for simplified Chinese characters
   - Chinese braille with tone
   - Two-cell Chinese braille
   - Taiwanese braille

DaveMielke commented:
[quoted lines by Bert Frees on 2019/08/22 at 15:53 -0700]

Hi:

Don't worry about subtle changes to improve things. I'm like that, too. :-)

>- Don't mention "forward" in either index-name or display-name. This is
>something that the user should never have to worry about. The program is
>always interested in either only forward translation, only backward
>translation, or both, so it should already filter out the unsuitable tables
>before presenting the list to the user. So, for example in da-dk-g16-lit.ctb,
>"Danish, uncontracted, 6-dot, forward" becomes "Danish, uncontracted, 6-dot"
>(index-name), and "Danish 6-dot uncontracted braille for forward translation"
>becomes "Danish 6-dot uncontracted braille" (display-name).

I aghree with you in principle but I added "forward" for a reason - to remove
duplicate index/display names. Each of them that now says "forward" has a
corresponding table that specifies #+direction:both and, in each case, removing
"forward" would result in pairs of identical index/display names. I dont' think
that duplicate names makes it easy for a user to know which table is which.

Now for what I really think: There are way too many Danish tables. Why do we
need a "both" and a "forward" version. Why not just get rid of one table from
each pair and then we wouldn't need to distinuish them from one another with
extra stuff.

>- You removed the "fully" from "fully contracted braille" (see for example
>da-dk-g26.ctb). I think I understand why you did this, but for me "contracted
>braille" also includes "partially contracted braille". I had added the "fully"
>to make it unambiguous. So my suggestion is to change it back to how it was
>before, and also add "fully" to the index-name. What do you think?

I guess it's a matter of taste. To me, "fully" is what one expects and most
will use while "partially" is an odd case tempoirarily used by those who are
learning braille. To me, therefore, saying "fully" adds extra baggage whereas
"partially" alerts the user that it's specially annotated to differentiate it
from what one expects to be the normal case.

English braille - at least EBAE - has a grade 1.5 which maybe someone should
write a table for. Should that mean, then, that the EBAE contracted table
should also have "fully" added to it? I think that'd look ugly.

>- In index-name (see for example da-dk-g26l.ctb) we could also maybe replace
>"partially contracted" with "contracted, partially"? Not sure about this
>though.

That doesn't read right in English. It implies that, in some way, the table is
only partially implemented. :-)

>- I was thinking about moving the "(UEB)" in "Unified English (UEB)
>uncontracted braille" to the end, because I find it a bit awkward in the
>middle. So it would become "Unified English uncontracted braille (UEB)". Same
>for EBAE. 

I udnerstand your reasoning, but that implies that UEB means uncontracted
unified braille. In other words, (something) implies that it defines everything
that precedes it. Why not just drop UEB, EBAE, etc from the names. I don't
think they're very helpful, actually.

>Moreover, if it's at the end it doesn't disturb as much, so we could possibly
>add similar abbreviations of braille codes and authorities elsewhere (BAUK,
>BFU, UIB, etc.). Or maybe this is going to create too much noise, I dunno. But
>in any case, it should be consistent. For example, adding it for EBAE also
>justifies adding it for BAUK.

Yes, I think those are noise and consistency should probably mean removing
them.

>- Just a tiny detail, but I would change "Esperanto, x-system" to "Esperanto
>x-system" (without the comma), because the "x-system" is part of the original
>writing system (with language code "eo-xsistemo") rather than a variant of the
>braille code. I know, very subtle difference, but still.

As I said above, I like getting things right even if the change is small. Sure,
go for it. I wasn't sure what to do with it because I didn't even know what it
meant.

>- For the French tables we could drop the "unified" because there are no
>French tables that are not "unified". 

Well, there used to be but removing them is fine because I doubt there'll be
any desire to implement the older tables as they're too Windows centric.

>Also I'm not sure whether it has the same meaning as the "U" in UEB, because
>then why are there separate tables for computer braille?

I suspect that it does. My guess is that it stands for something like "Braille
Français Unifié". But, still, sure, just remove Unified in the French case.

>- I would change "International Greek braille" to "Greek international
>braille" because the "international" has more affinity with "braille" then
>with "Greek".

But there's no such thing as international braille, and that's what such a
change would imply. I think, to make both of us happy, it should maybe be:

   Greek internationalized braille

>- I would change the index-name of no-no-8dot-fallback-6dot-g0.utb,
>"Norwegian, uncontracted, 8-dot, with 6-dot fallback", to "Norwegian,
>uncontracted, 8-dot, 6-dot fallback" which makes it slightly more compact
>which is important for index-name.

Yes, that's better.

>- You changed the display-name of all the tables for Mandarin to have the word
>"Mandarin" in it. While I think this is the right thing to do for index-name,
>it's not necessarily so for display-name. For example "Taiwanese Mandarin
>braille" just doesn't sound as good as "Taiwanese braille", and the latter is
>what is also used on the Wikipedia page.

Yes, removing Mandarin in the Taiwanese case makes sense and does look better.

>Wikipedia also talks about "(Mainland) Chinese braille", not "Mainland
>Mandarin braille". Besides "Mainland" without "China" doesn't even make that
>much sense, so I also suggest to replace "mainland" with "mainland China" in
>all the index-names. And for consistency in the index-names, I would also add
>"mainland China" to the index-name of the "2-cell" table.

Sure. My problem was that "Chinese" isn't a language, but "mainland China"
would be fine. 

>In summary, these are the names for the Mandarin tables I'm proposing:
>
>   index-name:
>   - Mandarin, mainland China, without tone
>   - Mandarin, mainland China, with tone
>   - Mandarin, mainland China, 2-cell
>   - Mandarin, Taiwan

Yes, except that I think "tone" should be "tones". Mandarin, after all, doesn't
just use neutral plus a single tone modifier - there are several. Saying "tone"
in the singular is probably the kind of English "mistake" that a Mandarin
speaker might make.
   
>   display-name:
>   - Chinese braille without tone, for simplified Chinese characters
>   - Chinese braille with tone
>   - Two-cell Chinese braille
>   - Taiwanese braille

Again, I think it should be "tones". Also, I really don't like saying Chinese
instead of Mandarin because Mandarin is the language. The people are Chinese
(as they also are in Hong Kong), but the language is Mandarin (as opposed to
Cantonese).

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Hi, some more comments:

> I aghree with you in principle but I added "forward" for a reason - to remove duplicate index/display names.

Hmm okay, I didn't notice that before. Still, my argument counts I think if we change the "both" to "backward". In this case the user will normally never have to see both tables in the list, so shouldn't get confused.

> Now for what I really think: There are way too many Danish tables. Why do we need a "both" and a "forward" version.

I wouldn't just delete the forward one, but rather change "both" into "backward". Then I'll ask Bue if there is a possibility to reduce the number of tables somehow, and if not if he could explain a bit more inside the tables what the differences are between the normal and "lit" versions.

The fact that there are many Danish tables is just because there are so many variants to choose from. I think we have to live with that.

> I guess it's a matter of taste. To me, "fully" is what one expects and most will use while "partially" is an odd case tempoirarily used by those who are learning braille. To me, therefore, saying "fully" adds extra baggage whereas "partially" alerts the user that it's specially annotated to differentiate it from what one expects to be the normal case.

I'm not sure it's like that in all countries though. But okay let's keep the change.

> Why not just drop UEB, EBAE, etc from the names. I don't think they're very helpful, actually.

OK sounds good.

> > - For the French tables we could drop the "unified" because there are no French tables that are not "unified". 
> 
> Well, there used to be but removing them is fine because I doubt there'll be any desire to implement the older tables as they're too Windows centric.
> 
> > Also I'm not sure whether it has the same meaning as the "U" in UEB, because then why are there separate tables for computer braille?
>
> I suspect that it does. My guess is that it stands for something like "Braille Français Unifié". But, still, sure, just remove Unified in the French case.

Yes, it does mean "unifié" but I thought maybe it had a different meaning then in UEB, because again why would you need a separate computer braille table then? It think it could become confusing for the user. Also I can find very little information about this on the web. It isn't mentioned very often so seems it has become not so relevant anymore. But anyway, I think we've already agreed that removing it is better :-)

> But there's no such thing as international braille, and that's what such a change would imply. I think, to make both of us happy, it should maybe be:
> 
>    Greek internationalized braille

OK that works for me. "International braille" does appear a lot in articles on the web though, so it seems to be common, but "internationalized" is fine too.

> > these are the names for the Mandarin tables I'm proposing:
> >
> >   index-name:
> >   - Mandarin, mainland China, without tone
> >   - Mandarin, mainland China, with tone
> >   - Mandarin, mainland China, 2-cell
> >   - Mandarin, Taiwan
> 
> Yes, except that I think "tone" should be "tones". Mandarin, after all, doesn't just use neutral plus a single tone modifier - there are several. Saying "tone" in the singular is probably the kind of English "mistake" that a Mandarin speaker might make.

I guess "tone" is something we could say in Dutch too. Maybe that's why I thought it is also valid English. It's one of those words that doesn't necessarily mean a single instance (and doesn't need an article in that case), but it can also mean a specific tone and can have an article and has a plural. But anyway you're the expert on the English language.

Maybe talking about "tone marking" is more accurate anyway?

> >   display-name:
> >   - Chinese braille without tone, for simplified Chinese characters
> >   - Chinese braille with tone
> >   - Two-cell Chinese braille
> >   - Taiwanese braille

> I really don't like saying Chinese instead of Mandarin because Mandarin is the language. The people are Chinese (as they also are in Hong Kong), but the language is Mandarin (as opposed to Cantonese).

Yeah sure, I know Chinese is not *the* language, but it is an adjective/country so there's no reason why a braille code can not be called like that. "Taiwanese" is also not a language.

Again to me it's more important that the names sound good and are recognizable by the users, so for example following Wikipedia is not a bad idea in this case because these pages are probably created by people who can know.

Besides I don't know if people from Hong Kong like to be called Chinese :-) so actually I think there can be little confusion. If there is, I would rather let someone who can know change the names.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/08/23 at 04:06 -0700]

Hi:

>> I aghree with you in principle but I added "forward" for a reason - to remove duplicate index/display names.
>
>Hmm okay, I didn't notice that before. Still, my argument counts I think if we
>change the "both" to "backward". In this case the user will normally never
>have to see both tables in the list, so shouldn't get confused.

In our case, we need forward and backward translation to ideally be precisely
inverse. Having to use two different tables - one for each direction - just
because it helps fix up a naming issue maybe is what we should be doing, but I
think I'd rather be using a single table that someone has verified to be
working correctly in both directions.

>> Now for what I really think: There are way too many Danish tables. Why do we need a "both" and a "forward" version.
>
>I wouldn't just delete the forward one, but rather change "both" into
>"backward". Then I'll ask Bue if there is a possibility to reduce the number
>of tables somehow, and if not if he could explain a bit more inside the tables
>what the differences are between the normal and "lit" versions.

Yes, so, at this point, especially given that the 3.11 release date is close, I
think we're left with this question. Is it better to introduce a lie by
changing both to backward or is it better to temporarily have a somewhat
awkward-looking name?

>> I suspect that it does. My guess is that it stands for something like
>"Braille Français Unifié". But, still, sure, just remove Unified in the French
>case.
>
>Yes, it does mean "unifié" but I thought maybe it had a different meaning then
>in UEB, because again why would you need a separate computer braille table
>then? 

Perhaps because French computer braille in different countries is (was)
different, just like US and UK English computer braille aren't the same.

>It think it could become confusing for the user. Also I can find very little
>information about this on the web. It isn't mentioned very often so seems it
>has become not so relevant anymore. But anyway, I think we've already agreed
>that removing it is better :-)

The French are very jealous about their language, even having an official
academy which strives to maintain their language's "purity". Given that, it
wouldn't surprise me at all if they even want to keep French computer braille
as "pure" as possible.

>> > these are the names for the Mandarin tables I'm proposing:
>> >
>> >   index-name:
>> >   - Mandarin, mainland China, without tone
>> >   - Mandarin, mainland China, with tone
>> >   - Mandarin, mainland China, 2-cell
>> >   - Mandarin, Taiwan
>> 
>> Yes, except that I think "tone" should be "tones". Mandarin, after all, doesn't just use neutral plus a single tone modifier - there are several. Saying "tone" in the singular is probably the kind of English "mistake" that a Mandarin speaker might make.
>
>I guess "tone" is something we could say in Dutch too. Maybe that's why I
>thought it is also valid English. It's one of those words that doesn't
>necessarily mean a single instance (and doesn't need an article in that case),
>but it can also mean a specific tone and can have an article and has a plural.
>But anyway you're the expert on the English language.
>
>Maybe talking about "tone marking" is more accurate anyway?

To me, that'd depend on the braille. Foe example, it'd be wrong to say French
braille with accent markings (notice I've used the plural here, too) since the
letters aren't merely annotated - completely different braille symbols are
used.

Back to "tone" (singular) for a moment. I, of course, knew the intention of the
word. To a casual reader, the phrase "Mandarin braille with tone" would mean
"Mandarin braille with expressiveness". That, of coruse, is entirely wrong
because modifying a Mandarin word with one of the four possible tones changes
the whole meaning of the word to something that isn't even related.

>Yeah sure, I know Chinese is not *the* language, but it is an
>adjective/country so there's no reason why a braille code can not be called
>like that. "Taiwanese" is also not a language.

>Besides I don't know if people from Hong Kong like to be called Chinese :-) so
>actually I think there can be little confusion. If there is, I would rather
>let someone who can know change the names.

Shall I ask on the list for Chinese people to answer this one?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
> Shall I ask on the list for Chinese people to answer this one?

OK sure, why not.

> I think I'd rather be using a single table that someone has verified to be working correctly in both directions.

In that case you should probably use only the "both" table (that is, automatically filter the "forward" one out), because that one is most likely to behave the same in forward and backward direction. In fact this is probably what you should be doing anyway, because you're only interested in tables that work both ways (de-g1.ctb and grc-international-en.utb are also marked "forward").

Note that the forward direction of one table and the backward direction of another table could in theory be more "in line" than the forward and backward direction of the same table. Using a single table doesn't guarantee anything about the back-translation capabilities. But let's forget about that fact for now...

I'm going to ask Bue's advice first. In fact, I'll invite him to the conversation.

Hi @BueVest. Dave an I are breaking our heads over the question how to present the list of available Danish tables to an end user. Do we need to name the tables that are optimized for forward translation (the "lit" tables) differently than the ones that work in both directions? Or can a program automatically filter out the tables that are not useful? In the latter case, and if the program needs both forward and backward directions, there are two possibilities: either use the "lit" table for forward translation and the other one for backward, or use the one table that can do both directions. I guess the real question is: what exactly is the added value of the "lit" tables? You probably told me before, but I forgot. It would be good to have a more elaborate explanation in the table.
